### PR TITLE
fix: do not panic while encoding oversized bodies

### LINKED ISSUE #1141: Returning body w/ >4GB data panics
### Version
`master` (`e1885211495e63d962bc1d00f9be6eeaab2bb901`)

### Description

This test panics with `thread 'codec::prost::tests::encode_big' panicked at 'assertion failed: len <= std::u32::MAX as usize', tonic/src/codec/encode.rs:127:5`, but instead the stream should probably return an error:

```rust

    #[tokio::test]
    async fn encode_big() {
        let encoder = MockEncoder::default();

        let msg = vec![0u8; u32::MAX as usize + 1];

        let messages = std::iter::once(Ok::<_, Status>(msg));
        let source = futures_util::stream::iter(messages);

        let body = encode_server(
            encoder,
            source,
            None,
            SingleMessageCompressionOverride::default(),
        );

        futures_util::pin_mut!(body);

        while let Some(r) = body.data().await {
            r.unwrap();
        }
    }
```

### PR DESCRIPTION
**Pull request by crepererum**
## Motivation
Fixes #1141.

## Solution
Replace assertion w/ an ordinary error path.

### CODE REVIEW

**Review by LucioFranco — APPROVED** [review]
LGTM this seems extremely logical and a good change: :D

File: tonic/src/codec/prost.rs [hunk 1]
```diff
@@ -136,6 +136,37 @@ mod tests {
         }
     }
 
+    #[tokio::test]
+    async fn encode_too_big() {
+        let encoder = MockEncoder::default();
+
+        let msg = vec![0u8; u32::MAX as usize + 1];
```
Comment (LucioFranco) [inline tonic/src/codec/prost.rs:143]: uh this seems like its failing on windows and I think that kinda makes sense when you try to allocate a vec of u32::MAX? Prob need to lower it for windows or something like that

Comment (crepererum) [reply tonic/src/codec/prost.rs:143]: W/o that much data, we're not hitting the limit that triggers the bug. I can exclude the test for windows though.

Comment (LucioFranco) [reply tonic/src/codec/prost.rs:143]: Yeah, lets see if that works, may just be some setting on windows that is making this fail.

Comment (crepererum) [reply tonic/src/codec/prost.rs:145]: done

**Pull request merged**